hazelcast/src/main/java/com/hazelcast/config/DataConnectionConfig.java

```diff
@@ -181,6 +184,26 @@ public DataConnectionConfig setProperties(Properties properties) {
         return this;
     }
 
+    /**
+     * Checks if this configuration object is in coherent state - has all required properties set.
+     */
+    @PrivateApi
+    public void validate() {
+        List<String> errors = new ArrayList<>();
+        if (name == null || name.isEmpty()) {
+            errors.add("Data connection name must be non-null and contain text");
+        }
+        if (type == null || type.isEmpty()) {
+            errors.add("Data connection type must be non-null and contain text");
+        }
+        if (properties == null) {
```

Comment:
`properties == null` can't happen

Reply:
A bit paranoic check ;) If someone edited via reflection for whatever reason (e.g. serialization framework)